homeassistant/components/mqtt/alarm_control_panel.py

```diff
@@ -89,7 +89,7 @@
             CONF_COMMAND_TEMPLATE, default=DEFAULT_COMMAND_TEMPLATE
         ): cv.template,
         vol.Required(CONF_COMMAND_TOPIC): valid_publish_topic,
-        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Optional(CONF_NAME): cv.string,
```

Comment:
Since we're now introducing modern naming, I think we should also allow explicitly setting the name to `None` to mark the primary entity:
```suggestion
        vol.Optional(CONF_NAME): volAny(None, cv.string),
```

Reply:
I'll will add a test for this as well